Console/retro/Console.cc

```diff
@@ -136,11 +206,21 @@ void Console::DrawCells(short x1, short x2, short y, bool erase)
             cursorDrawn = false;
         }
     }
-    
+
     if(erase)
         EraseRect(&r);
     MoveTo(r.left, r.bottom - 2);
-    DrawText(&chars[y * cols + x1], 0, x2 - x1);
+
+    Attributes a=chars[y * cols + x1].attrs;
+    SetAttributes(a);
+    for(int i=x1; i<x2; ++i)
```

Comment:
This is will be noticeably slower than `DrawText` on old machines.
I have some ideas about how to optimize it - basically, combining consecutive chars with the same attributes into bigger `DrawText` calls. I'll try that out when I have the time.
It should be OK as is for the meantime.

Reply:
I thought that, too. Probably something similar to the old code I wrote before putting together text and attributes. Here is a chuck from https://github.com/DarwinNE/Retro68/commit/0df7517c16adbc0d78ed5057ec3a4c59c6ee2d36

~~~~
    a=attrs[y * cols + x1];
    SetAttributes(a);
    start=x1;
    for(int i=x1; i<x2; ++i)
    {
        if(a!=attrs[y * cols + i])
        {
            DrawText(&chars[y * cols + start], 0, i - start);
            a=attrs[y * cols + i];
            SetAttributes(a);
            start=i;
        }
    }
    DrawText(&chars[y * cols + start], 0, x2 - start);
~~~~

But with the new code, one would need to copy the message in a buffer.